lnwire/dyn_propose.go

```diff
@@ -0,0 +1,249 @@
+
+	// DPMaxAcceptedHtlcs is the TLV type number that identifies the record
+	// for DynPropose.MaxAcceptedHTLCs.
+	DPMaxAcceptedHtlcs tlv.Type = 10
+
+	// DPFundingPubkey is the TLV type number that identifies the record for
+	// DynPropose.FundingKey.
+	DPFundingPubkey tlv.Type = 12
+
+	// DPChannelType is the TLV type number that identifies the record for
+	// DynPropose.ChannelType.
+	DPChannelType tlv.Type = 14
+)
+
+// DynPropose is a message that is sent during a dynamic commitments negotiation
+// process. It is sent by both parties to propose new channel parameters.
+type DynPropose struct {
+	// ChanID identifies the channel whose parameters we are trying to
+	// re-negotiate.
+	ChanID ChannelID
+
+	// DustLimit, if not nil, proposes a change to the dust_limit_satoshis
+	// for the sender's commitment transaction.
+	DustLimit *btcutil.Amount
+
+	// MaxValueInFlight, if not nil, proposes a change to the
+	// max_htlc_value_in_flight_msat limit of the sender.
+	MaxValueInFlight *MilliSatoshi
+
+	// ChannelReserve, if not nil, proposes a change to the
+	// channel_reserve_satoshis requirement of the recipient.
+	ChannelReserve *btcutil.Amount
+
+	// HtlcMinimum, if not nil, proposes a change to the htlc_minimum_msat
+	// limit of the sender.
+	HtlcMinimum *MilliSatoshi
+
+	// CsvDelay, if not nil, proposes a change to the to_self_delay
+	// requirement of the recipient.
+	CsvDelay *uint16
+
+	// MaxAcceptedHTLCs, if not nil, proposes a change to the
+	// max_accepted_htlcs limit of the sender.
+	MaxAcceptedHTLCs *uint16
+
+	// FundingKey, if not nil, proposes a change to the funding_pubkey
+	// parameter of the sender.
+	FundingKey *btcec.PublicKey
+
+	// ChannelType, if not nil, proposes a change to the channel_type
+	// parameter.
+	ChannelType *ChannelType
+}
+
+// A compile time check to ensure DynPropose implements the lnwire.Message
+// interface.
+var _ Message = (*DynPropose)(nil)
+
+// Encode serializes the target DynPropose into the passed io.Writer.
+// Serialization will observe the rules defined by the passed protocol version.
+//
+// This is a part of the lnwire.Message interface.
+func (dp *DynPropose) Encode(w *bytes.Buffer, _ uint32) error {
+	if err := WriteChannelID(w, dp.ChanID); err != nil {
+		return err
+	}
+
+	tlvRecords := []tlv.Record{}
+	if dp.DustLimit != nil {
+		tlvRecords = append(tlvRecords, tlv.MakePrimitiveRecord(
+			DPDustLimitSatoshis, dp.DustLimit))
+	}
+	if dp.MaxValueInFlight != nil {
+		tlvRecords = append(tlvRecords, tlv.MakePrimitiveRecord(
+			DPMaxHtlcValueInFlightMsat, dp.MaxValueInFlight))
+	}
+	if dp.ChannelReserve != nil {
+		tlvRecords = append(tlvRecords, tlv.MakePrimitiveRecord(
+			DPChannelReserveSatoshis, dp.ChannelReserve))
+	}
+	if dp.HtlcMinimum != nil {
+		tlvRecords = append(tlvRecords, tlv.MakePrimitiveRecord(
+			DPHtlcMinimumMsat, dp.HtlcMinimum))
+	}
+	if dp.CsvDelay != nil {
+		tlvRecords = append(tlvRecords, tlv.MakePrimitiveRecord(
+			DPToSelfDelay, dp.CsvDelay))
```

Comment:
Style nit: we usually have a trailing comma then parens for cases like this.

Reply:
Any chance you could suggest the change you want here? Is the comma supposed to go after `dp.CsvDelay` and the parens on the next line?

Reply:
I think we usually do something like:
```go
tlvRecords = append(
     tlvRecords, tlv.MakePrimitiveRecord(
          DPToSelfDelay, dp.CsvDelay,
     ),
)
```

alternatively, you could move the call to `MakePrimitiveRecord` above the `append` so there aren't as many line breaks in the parenthesis